instrumentation/netty/netty-3.8/src/main/java/io/opentelemetry/auto/instrumentation/netty/v3_8/client/HttpClientRequestTracingHandler.java

```diff
@@ -58,15 +62,20 @@ public void writeRequested(final ChannelHandlerContext ctx, final MessageEvent m
     channelTraceContext.setClientParentSpan(TRACER.getCurrentSpan());
 
     HttpRequest request = (HttpRequest) msg.getMessage();
+
     Span span = TRACER.startSpan(request);
-    TRACER.onPeerConnection(span, (InetSocketAddress) ctx.getChannel().getRemoteAddress());
+    try (Scope scope = currentContextWith(span)) {
+      TRACER.onPeerConnection(span, (InetSocketAddress) ctx.getChannel().getRemoteAddress());
+      Context context = withSpan(span, Context.current());
+      OpenTelemetry.getPropagators().getHttpTextFormat().inject(context, request.headers(), SETTER);
 
-    try (Scope scope = TRACER.startScope(span, request)) {
       channelTraceContext.setClientSpan(span);
-      ctx.sendDownstream(msg);
-    } catch (final Throwable throwable) {
-      TRACER.endExceptionally(span, throwable);
-      throw throwable;
+      try {
```

Comment:
i think u can remove this nested try, e.g.

```
    Span span = TRACER.startSpan(request);
    TRACER.onPeerConnection(span, (InetSocketAddress) ctx.getChannel().getRemoteAddress());
    Context context = withSpan(span, Context.current());
    OpenTelemetry.getPropagators().getHttpTextFormat().inject(context, request.headers(), SETTER);

    channelTraceContext.setClientSpan(span);

    try (Scope scope = currentContextWith(span)) {
      ctx.sendDownstream(msg);
    } catch (final Throwable throwable) {
      TRACER.endExceptionally(span, throwable);
      throw throwable;
    } finally {
      if (parentScope != null) {
        parentScope.close();
      }
    }
```

Reply:
channelTraceContext.setClientSpan(span); 

does this require scope?

Reply:
no, because it's only calling our internal code